src/main/java/com/mackervoy/calum/mud/behaviour/task/TaskActorFactory.java

```diff
@@ -0,0 +1,38 @@
+package com.mackervoy.calum.mud.behaviour.task;
+
+import java.lang.reflect.InvocationTargetException;
+import java.util.Dictionary;
+import java.util.Hashtable;
+
+/**
+ * @author Calum Mackervoy
+ * Resolves at runtime a TaskActor class for a requested RDF class
+ */
+public class TaskActorFactory {
+	private static Dictionary<String, Class<? extends ITaskActor>> taskActors = 
+			new Hashtable<String, Class<? extends ITaskActor>>();
+	
+	/*
+	 * Registers a provider to an RDF type with String
+	 * @returns true if the object could be registered, false if another class has already registered
+	 */
+	public static boolean register(String rdfType, Class<? extends ITaskActor> actor) {
+		if(taskActors.put(rdfType, actor) == null) {
+			return true;
+		}
+		return false;
+	}
+	
+	public ITaskActor getActor(String rdfType) {
```

Comment:
I think we could use an Optional here, to make clear that `getActor(...)` isn't guaranteed to return a ITaskActor